memiavl/db.go

```diff
@@ -372,18 +404,33 @@ func (db *DB) reloadMultiTree(mtree *MultiTree) error {
 	return nil
 }
 
+// rewriteIfApplicable execute the snapshot rewrite strategy according to current height
+func (db *DB) rewriteIfApplicable(height int64) {
+	if height%int64(db.snapshotInterval) != 0 {
+		return
+	}
+
+	if err := db.rewriteSnapshotBackground(); err != nil {
+		db.logger.Error("failed to rewrite snapshot in background", "err", err)
+	}
+}
+
 type snapshotResult struct {
 	mtree   *MultiTree
 	err     error
 	version uint32
 }
 
-// RewriteSnapshotBackground rewrite snapshot in a background goroutine,
-// `Commit` will check the complete status, and switch to the new snapshot.
 func (db *DB) RewriteSnapshotBackground() error {
 	db.mtx.Lock()
 	defer db.mtx.Unlock()
 
+	return db.rewriteSnapshotBackground()
+}
+
+// rewriteSnapshotBackground rewrite snapshot in a background goroutine,
+// `Commit` will check the complete status, and switch to the new snapshot.
+func (db *DB) rewriteSnapshotBackground() error {
```

Comment:
seems hash get mutate when rewriteSnapshotBackground

Reply:
after switch the memiavl store need to reload.